xarray/tests/test_backends.py

```diff
@@ -810,6 +810,17 @@ def equals_latlon(obj):
                 assert "coordinates" not in ds["lat"].attrs
                 assert "coordinates" not in ds["lon"].attrs
 
+        original["temp"].encoding["coordinates"] = "lat"
+        with self.roundtrip(original) as actual:
+            assert_identical(actual, original)
+        with create_tmp_file() as tmp_file:
+            original.to_netcdf(tmp_file)
+            with open_dataset(tmp_file, decode_coords=False) as ds:
+                assert equals_latlon(ds.precip.attrs["coordinates"])
+                assert "lon" not in ds["temp"].attrs["coordinates"]
+                assert "coordinates" not in ds["lat"].attrs
+                assert "coordinates" not in ds["lon"].attrs
```

Comment:
Am I correct in my interpretation that this should also work?

```python
assert 'lon' not in ds['temp'].coords
```

Reply:
no. `"lon"` is still a coord in xarray-land because `ds["precip"].attrs["coordinates"} = "lat lon"` when the dataset is read. Other software might do the right thing.